### PR TITLE
Final bug fixes

### LINKED ISSUE #173: [PE-D][Tester E] Duplicate nc/NOTE_CONTENT in adding note causes unexpected behaviour
How to recreate:

1. Enter command: `note 1 nt/title nc/content_one nc/content_two`

Expected outcome: adding note will be caused an error due to duplicate nc/ property.
Actual outcome: 1 note added, title: "title" with content: "content_two".

Problem:
Confusing to the users as they may expect "content_one" to be shown.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: xplus2g4/ped#1

### LINKED ISSUE #174: [PE-D][Tester E] Duplicate nt/TITLE in adding note causes unexpected behaviour
How to recreate:

1. Enter command: `note 1 nt/title_one nt/title_two nc/content`

Expected outcome: adding note will be caused an error due to duplicate nt/ property.
Actual outcome: 1 note added, title: "title_two" with content: "content".

Problem:
Confusing to the users as they may expect "title_one" to be shown.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: xplus2g4/ped#2

### LINKED ISSUE #176: [PE-D][Tester D] Add command accepts invalid phone numbers
I had expected that adding a patient with an invalid phone number (<8 or >8) would be unsuccessful. 

However, the actual adding of patients with invalid phone numbers was successful.

This might be a little problematic when it comes to situations where users might miss out a digit or add an extra digit when adding in a new patient. 
While this might be rare and might possibly rectified with the edit feature, this mistake is not made known to the user unless the user double checks the added patient especially if they might be adding multiple patients at once.

A suggestion for this would be to prevent users from adding a patient with an invalid phone number. An alternative would be to add a warning message after the patient had been added successfully to warn the user that the added patients phone number was invalid.

The following inputs and outputs were what I did and received to view this issue.

Input (phone number <8 digits): 
add n/ppp p/123456 a/p
Output:

Input (phone number >8 digits):
add n/pppp p/123456789012345 a/p
Output:

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: botosek/ped#1

### LINKED ISSUE #179: [PE-D][Tester B] Edit command does not support tags of different cases
In the add command, I can add same tags with different letters capitalised, but in the edit command, this feature is not present.

Steps to reproduce:
Add a new user with the tags 
`add n/user123 p/123 a/addr t/tag t/TAG`

Edit the newly added user:
`edit 1 t/tag t/TAG`

The expected result should not have changed from the add command, however, it only added tag and not TAG.

This could be a problem if there are acronyms that spell out the same way, but might have different cases.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: RT0118/ped#1

### LINKED ISSUE #191: [PE-D][Tester E] Duplicate names not allowed in the app
For a patient app, it might be quite common to see patients with duplicate names. It will be a headache to the psychiatrist that he cannot add patients with duplicate names

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: xplus2g4/ped#9

### LINKED ISSUE #193: [PE-D][Tester B] Note title with different cases not accepted
When adding 2 notes with similar names, but different cases, the application will reject the 2nd note to be added.

Steps to reprocuce:

Add the 1st note to a patient
`note 1 nt/ABC - 1 nc/something related to patient's ABC problem`

Add the 2nd note to the same patient
`note 1 nt/abc - 1 nc/something related to patient's abc problem`

The expected result is for both notes to be accepted, but it shows an error instead about duplicate notes.

In the UG, it is not specified that note titles are case-insensitive, so it is assumed that the notes can be added. Also, if a user needs to add notes where the title contains acronyms of different things with the same lettering, it will not work for them.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: RT0118/ped#4

### LINKED ISSUE #200: [PE-D][Tester A] edit function should be following displayed list
as it is stated that the index should be following displayed list, may be it should be stated in userguide too?

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: Alolononon/ped#8

### LINKED ISSUE #204: [PE-D][Tester A] "note" command - userguide should state that the INDEX should be refering to displayed list?
as note command is performed, the INDEX is following the displayed list. it is stated in app but it is not stated userguide. 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: Alolononon/ped#10

### LINKED ISSUE #217: [PE-D][Tester C] Standardisation of Expected Error Message in DG
## Description:
Some error messages doesn’t match the expected error message in the developer guide. In the case of Deleting an invalid patient index it matches but in the case of adding notes to an invalid patient index doesn’t match. 

## Suggestion:
Suggestion would be if the error message is too long, could put the important part of the error message and put … wherever needed. Standardised expected error message in DG. There is multiple instances of this documentation issue.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: TheEnd-alr-taken/ped#8

### LINKED ISSUE #218: [PE-D][Tester C] Standardized putting Expected Outcome starting on a new line.
There are a few instances of this

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: TheEnd-alr-taken/ped#9

### LINKED ISSUE #220: Add to blue box (UG): Patient index has to be within list of patients displayed, and going too big will cause different error messages

### PR DESCRIPTION
**Pull request by benedictleejr**
Resolves #173, resolves #174, resolves #179, resolves #191, resolves #193, resolves #217 
Resolves #218, resolves #220 
Resolves #176, resolves #200, resolves #204

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -606,35 +606,58 @@ Prerequisites for all patient management commands:
...
+Expected outcome:
+```bash
+New patient added: John Doe; Phone: 98765432; Address: 311, Clementi Ave 2, #02-25; 
+Notes: ; Tags: [Schizophrenia][Anxiety]
+```
+<br>
+
+Test case: `add n/John Doe p/123 a/Address` (Testing duplicate patient)<br>
+Expected outcome: No patient is added.<br>
+Expected error message: `This patient already exists in the app. Duplicate patient name is not allowed`.
```
Comment (Golddirio) [inline docs/DeveloperGuide.md:623]: Good to have duplicate name case.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -606,35 +606,58 @@ Prerequisites for all patient management commands:
...
+Expected outcome:
+```bash
+Edited Patient: John Smith; Phone: 87438807; Address: Blk 30 Geylang Street 29, #06-40; 
+Notes: ; Tags: [friends]
+```
+<br>
+
+Test case: `edit 999 n/John Smith` (Testing invalid index)<br>
+Expected outcome: No patient is edited.<br>
+Expected error message: `Invalid index! Please provide a positive integer within the patient list range!`
```
Comment (Golddirio) [inline docs/DeveloperGuide.md:639]: Good to have invalid index case.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -644,83 +667,128 @@ Prerequisites for all note management commands:
...
 
-Test case: `filternote 1 nt/Visit`  
-Expected outcome: Notes of patient at index 1 containing "Visit" in their titles are displayed.
+Test case: `filternote 1 nt/Visit`<br>
+Expected outcome: Notes of patient at index 1 containing "Visit" in their titles are displayed.<br>
+Expected output: `Displaying notes. See notes panel below.`
+
+Test case: `filternote 1 nt/Nonexistent` (Filtering non-existent note title)<br>
+Expected outcome: Error details shown in the status message.<br>
+Expected error message: `Note Title does not exist: Nonexistent`
```
Comment (Golddirio) [inline docs/DeveloperGuide.md:720]: I like the special case details.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -218,8 +222,16 @@ add n/John Doe p/98005442 a/123 Clementi Street t/ADHD
...
+⚠️ **Warning:** Duplicate patient names are not allowed, for clarity.
+
+💡 **Tip:** 
+Use **tags** (`t/`) to categorize patients (e.g., `t/Anxiety`). Duplicate tags (case-insensitive) will be auto-filtered to only preserve the first instance seen.
+
+**Example:**
+```bash
+add n/John Doe p/98005442 a/123 Clementi Street t/ADHD t/adhd t/Anxiety t/ANXIETY
+```
+In this example, only the tags `ADHD` and `Anxiety` will be preserved.
```
Comment (Golddirio) [inline docs/UserGuide.md:234]: Good example provided!

File: docs/UserGuide.md [hunk 5]
```diff
@@ -340,7 +357,8 @@ filternote 1 nt/23
 <div style="page-break-after: always;"></div>
 
 ### **Adding a Note** {#adding-a-note}
-You can add notes to a patient to keep track of discussions, diagnoses, etc.
+You can add notes to a patient to keep track of discussions, diagnoses, etc.<br>
+Duplicate note titles (case-insensitive) are not allowed.
```
Comment (Golddirio) [inline docs/UserGuide.md:361]: Good to remind that no duplicate titles are allowed.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -392,7 +410,8 @@ View the changes after:
   <img src="images/viewAfterEditNoteCommand.png" alt="image" width="800"/>
 
 
-💡 **Tip:** You can edit only the necessary fields without altering other details.
+💡 **Tip:** You can only edit the **note content**. If you wish to change a note title, you can [delete a note](#deleting-a-note) and [add a new note](#adding-a-note) with the changed title!
+
```
Comment (Golddirio) [inline docs/UserGuide.md:414]: I like this improved tip!

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 7]
```diff
@@ -42,14 +45,36 @@ public AddCommand parse(String args) throws ParseException {
...
+        for (Tag tag : rawTagList) {
+            String lowercaseTagName = tag.tagName.toLowerCase();
+            if (!uniqueTags.containsKey(lowercaseTagName)) {
+                uniqueTags.put(lowercaseTagName, tag);
+            }
+        }
+
+        return new HashSet<>(uniqueTags.values());
+    }
+
```
Comment (Golddirio) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:77]: Good to have this filterTags method. Look like free of bugs. Good job!

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 8]
```diff
@@ -203,4 +190,94 @@ public void parse_invalidValue_failure() {
...
+
+        rawTags.addAll(Arrays.asList(adhd, depression, anxiety));
+
+        Set<Tag> filteredTags = parser.testFilterTags(rawTags);
+
+        assertEquals(3, filteredTags.size(), "Should keep all tags when there are no duplicates");
+        assertTrue(filteredTags.contains(adhd), "Should contain the ADHD tag");
+        assertTrue(filteredTags.contains(depression), "Should contain the Depression tag");
+        assertTrue(filteredTags.contains(anxiety), "Should contain the Anxiety tag");
+    }
```
Comment (Golddirio) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:282]: Very diligent testing of filterTags. Covers most test cases. Look good!

**Review by Golddirio — COMMENTED** [review]
LGTM as all the updates are valid and correct!
